File: CustomPopup/app/src/main/java/com/example/nabha/custompopup/activity/MainActivity.java
package com.example.nabha.custompopup.activity;

import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.example.nabha.custompopup.R;
import com.example.nabha.custompopup.fragments.CustomDialogFragment;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    private static final String TAG = MainActivity.class.getSimpleName();
    private Button mShowDialogButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mShowDialogButton = findViewById(R.id.showDialog);
        mShowDialogButton.setOnClickListener(this);
    }


    @Override
    public void onClick(View view) {
        Log.i(TAG, "Dialog!!! ");

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        Fragment prev = getSupportFragmentManager().findFragmentByTag("dialog");
        if (prev != null) {
            ft.remove(prev);
        }
        ft.addToBackStack(null);
        DialogFragment dialogFragment = new CustomDialogFragment();
        dialogFragment.show(ft, "dialog");
    }
}

File: KBuilder2/app/src/main/java/com/example/kbuilder2/activity/MainActivity.kt
package com.example.kbuilder2.activity

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.example.kbuilder2.R
import com.example.kbuilder2.model.Attendant
import com.example.kbuilder2.model.NonVegMealBuilder
import com.example.kbuilder2.model.VegMealBuilder

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val vegMealBuilder = VegMealBuilder()
        var attendant = Attendant(vegMealBuilder)
        attendant.constructMeal()
        Log.i("meal", attendant.getMeal().toString())

        val nonVegMealBuilder = NonVegMealBuilder()
        attendant = Attendant(nonVegMealBuilder)
        attendant.constructMeal()
        Log.i("meal", attendant.getMeal().toString())
    }
}
